### PR TITLE
update rio-tiler and refactor overviews

### PR DESCRIPTION
**Pull request by vincentsarago**
This PR does:
- Update for rio-tiler 2.0b2 (COGReader)
- add changes from #91 (using in-memory/temp file)
- add `merge` cli to create high resolution image from a mosaicJSON 🤷‍♂️

cc @kylebarron

### CODE REVIEW

File: cogeo_mosaic/overviews.py [hunk 1]
```diff
@@ -22,52 +25,51 @@
...
 
 
-def _get_blocks(extrema):
+def _get_blocks(extrema: Dict, tilesize: int = 256):
     for j in range(extrema["y"]["max"] - extrema["y"]["min"]):
         row = j * 256
         for i in range(extrema["x"]["max"] - extrema["x"]["min"]):
             col = i * 256
+
             yield (j, i), Window(col_off=col, row_off=row, width=256, height=256)
```
Comment (vincentsarago) [inline cogeo_mosaic/overviews.py:73]: missing tilesize

File: cogeo_mosaic/overviews.py [hunk 2]
```diff
@@ -76,8 +78,10 @@ def create_low_level_cogs(
     output_profile: Dict,
     prefix: str = "mosaic_ovr",
     max_overview_level: int = 6,
+    method: str = "first",
```
Comment (vincentsarago) [inline cogeo_mosaic/overviews.py:81]: we can now choose what method to use 🤷‍♂️

File: cogeo_mosaic/overviews.py [hunk 3]
```diff
@@ -124,80 +133,200 @@ def create_low_level_cogs(
...
+                min_tile_size = tilesize = min(
+                    int(output_profile["blockxsize"]),
+                    int(output_profile["blockysize"]),
+                )
+                overview_level = get_maximum_overview_level(
+                    tmp_dst.width, tmp_dst.height, minsize=min_tile_size
+                )
+                overviews = [2 ** j for j in range(1, overview_level + 1)]
+                tmp_dst.build_overviews(overviews)
+                copy(tmp_dst, output_path, copy_src_overviews=True, **params)
```
Comment (vincentsarago) [inline cogeo_mosaic/overviews.py:332]: Need to add test for this function

File: cogeo_mosaic/overviews.py [hunk 4]
```diff
@@ -124,80 +133,200 @@ def create_low_level_cogs(
...
+    mosaic_path: str,
+    output_path: str,
+    output_profile: Dict,
+    method: str = "first",
+    config: Dict = None,
+    threads=1,
+    in_memory: bool = True,
+) -> None:
+    """
+    Create WebOptimized Overview COG from a mosaic definition file.
```
Comment (kylebarron) [inline cogeo_mosaic/overviews.py:227]: I'd suggest a better docstring for what the difference between `high_level` and `low_level` is

File: cogeo_mosaic/scripts/cli.py [hunk 5]
```diff
@@ -321,8 +341,67 @@ def overview(
         output_profile,
         prefix,
         max_overview_level=overview_level,
+        method=method,
+        config=config,
+        threads=threads,
+        in_memory=in_memory,
+    )
+
+
```
Comment (kylebarron) [inline cogeo_mosaic/scripts/cli.py:350]: what?? `overview` now does nothing... It creates the config but doesn't use it for anything.

**Comment by kylebarron** [comment]
How does `merge` differ from `overview`?

> * add `merge` cli to create high resolution image from a mosaicJSON 🤷‍♂️

That's not what I'd expect from `cogeo-mosaic merge`... I would've expected `merge` to combine multiple MosaicJSONs..

> * add changes from #91 (using in-memory/temp file)

To be clear I never succeeded in that PR in low-memory use by iterating over windows. When I used it I still ended up having huge amounts of data stored in memory at once. You could see that the output file held 0 bytes until the entire process was finished.

File: cogeo_mosaic/scripts/cli.py (hunk 5 above)
Comment (vincentsarago) [reply cogeo_mosaic/scripts/cli.py:350]: I don't get what you are saying here @kylebarron

**Comment by vincentsarago** [comment]
@kylebarron 
> To be clear I never succeeded in that PR in low-memory use by iterating over windows. When I used it I still ended up having huge amounts of data stored in memory at once. You could see that the output file held 0 bytes until the entire process was finished.

This might be due to GDAL Cache, you can try setting GDAL_CACHEMAX really low and see how it goes

**Comment by vincentsarago** [comment]
> That's not what I'd expect from cogeo-mosaic merge... I would've expected merge to combine multiple MosaicJSONs..

renamed the cmd to `mosaic-to-cog`

File: cogeo_mosaic/scripts/cli.py [hunk 6]
```diff
@@ -14,7 +14,11 @@
 from cogeo_mosaic import version as cogeo_mosaic_version
 from cogeo_mosaic.backends import MosaicBackend
 from cogeo_mosaic.mosaic import MosaicJSON
-from cogeo_mosaic.overviews import create_low_level_cogs
+from cogeo_mosaic.translate import (
```
Comment (vincentsarago) [inline cogeo_mosaic/scripts/cli.py:17]: renamed the submodule

**Comment by kylebarron** [comment]
> renamed the cmd to `mosaic-to-cog`

you don't like just `overview`?

**Comment by vincentsarago** [comment]
this is replacing the `merge` command @kylebarron

**Pull request closed without merging**